File: token/token.go
package token

type TokenType string

type Token struct {
	Type    TokenType
	Literal string
}

var keywords = map[string]TokenType{
	"func":   FUNCTION,
	"let":    LET,
	"if":     IF,
	"else":   ELSE,
	"return": RETURN,
}

const (
	ILLEGAL = "illegal"
	EOF     = "eof"

	// Identifiers + literals
	IDENT = "ident"
	INT   = "int"

	// Operators
	ASSIGN   = "="
	PLUS     = "+"
	MINUS    = "-"
	BANG     = "!"
	ASTERISK = "*"
	SLASH    = "/"
	LT       = "<"
	GT       = ">"
	EQ       = "=="
	NOT_EQ   = "!="

	// Delimiters
	COMMA     = ","
	SEMICOLON = ";"
	LPAREN    = "("
	RPAREN    = ")"
	LBRACE    = "{"
	RBRACE    = "}"

	// Keywords
	FUNCTION = "FUNCTION"
	LET      = "LET"
	TRUE     = "TRUE"
	FALSE    = "FALSE"
	IF       = "IF"
	ELSE     = "else"
	RETURN   = "RETURN"
)

func NewToken(token TokenType, literal string) *Token {
	return &Token{token, literal}
}

func LookupIdent(ident string) TokenType {
	val, ok := keywords[ident]
	if ok {
		return val
	}

	return IDENT
}

File: lexer/src/token.rs
#[derive(Debug, PartialEq)]
pub enum Token {
    Illegal,
    EOF,
    Identifier { name: String },
    Integer(i32),
    Str(String),

    // Operators
    Plus,
    Minus,
    Slash,
    Asterisk,
    Modulo,

    // Symbols
    Assign,
    Equal,
    NotEqual,
    Bang,
    LeftParen,
    RightParen,
    LeftBrace,
    RightBrace,
    LeftBracket,
    RightBracket,
    Comma,
    Hashtag,
    DoubleQuote,
    Pipe,
    Semicolon,

    // Keywords
    Function,
    Match,
    If, 
    Else,
    Return,
    For,
    Break,
    Continue,
}
    